src/DataDepsGenerators.jl

```diff
@@ -125,14 +174,57 @@ function format_meta(data::Any, label="")
     return "\n" * indent(label * string(data))
 end
 
-function generate(repo::DataRepo,
-                  dataname,
-                  shortname = nothing
-    )
 
-    meta = find_metadata(repo, dataname, shortname)
+# function generate(repo::DataRepo,
+#                   dataname,
+#                   shortname = nothing
+#     )
+#     generators = [CKAN(), DataCite(), Figshare(), JSONLD_DOI(), JSONLD_Web()]
+#     metadatas = []
+#     push!(generators, repo)
+#     for ii in generators
+#         meta = find_metadata(ii, dataname, shortname)
+#         if !isa(meta, Void)
+#             push!(metadatas, meta)
+#         end
+#     end
+#     aggregate(metadatas)
+# end
+
+function generate(repo::DataRepo, dataname; kwargs...)
+    generate([repo], dataname; kwargs...)
+end
+
+function generate(dataname; kwargs...)
+    all_repos = [T() for T in leaf_subtypes(DataRepo)]
+    generate(all_repos , dataname; kwargs...)
+end
 
-    body(meta)
+function generate(repos, dataname, shortname = nothing, show_failures=false)
+	retrieved_metadatas = []
+	failures = Tuple{DataRepo, Exception}[]
+	
+	# Get all the metadata we can
+	@sync for repo in repos
+		@async try
+			metadata = find_metadata(repo, dataname, shortname)
+			push!(retrieved_metadatas, metadata)
+		catch err
+			push!(failures, (repo, err))
+		end
+	end
+	
+	# Displace errors if required
+	if length(retrieved_metadatas) == 0 || show_failures
+		for (repo, err) in failures
+			println(repo, " failed due to")
+			println(err)
+			println()
+		end
+	end
+	
+	# merge them
+	aggregate(retrieved_metadatas)
 end
```

Comment:
Indentation in this section has been done with tabs.
Everything else is indented with spaces.

I suspect this is because the code I gave you in the comment was that way.
There should be a `retab`  or `expandtabs` or similar command in your editor that will fix this.

Reply:
sorted now